src/differentials/composite.jl

```diff
@@ -180,6 +180,11 @@ canonicalize(comp::Composite{<:Tuple, <:Tuple}) = comp
 # Dict composite are always in their canonical form.
 canonicalize(comp::Composite{<:Any, <:AbstractDict}) = comp
 
+# Composites of unspecified primal types
```

Comment:
```suggestion
# Composites of unspecified primal types (indicated by specifying exactly `Any`)
# all combinations of type-params are specified here to avoid ambiguities
```